contracts/core/PolicyEngine.sol

```diff
@@ -154,11 +154,13 @@ abstract contract PolicyEngine is IPolicyEngine {
     ) internal {
         // Creating access selector for a policy
         AccessSelector.T access = AccessSelector.create(target, selector, operation);
-        $policies[safe][access] = policy;
 
         // Configuring policy
         require(IPolicy(policy).configure(safe, access, data), PolicyConfigurationFailed());
 
+        // Update the policy mapping
+        $policies[safe][access] = policy;
```

Comment:
There is a potential reentrency issue here right? Shouldn't we set storage before calling the policy?